app.go

```diff
@@ -0,0 +1,423 @@
+		}
+		start_date_since  = v["start_date_since"]
+		start_date_before = v["start_date_before"]
+		end_date_since    = v["end_date_since"]
+		end_date_before   = v["end_date_before"]
+		null_end          = v["include_null_end"]
+	)
+
+	if len(null_end) > 0 {
+		filters.IncludeNullEnd = true
+	}
+
+	for _, startdate := range start_date_since {
+		parsed, err := time.Parse(time.RFC3339Nano, startdate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.StartDateSince = append(filters.StartDateSince, parsed)
+	}
+
+	for _, startdate := range start_date_before {
+		parsed, err := time.Parse(time.RFC3339Nano, startdate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.StartDateBefore = append(filters.StartDateBefore, parsed)
+	}
+
+	for _, enddate := range end_date_since {
+		parsed, err := time.Parse(time.RFC3339Nano, enddate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.EndDateSince = append(filters.EndDateSince, parsed)
+	}
+
+	for _, enddate := range end_date_before {
+		parsed, err := time.Parse(time.RFC3339Nano, enddate)
+		if err != nil {
+			errored(writer, err.Error())
+			return
+		}
+		filters.EndDateBefore = append(filters.EndDateBefore, parsed)
+	}
+
+	tx, err := a.db.BeginTx(context.TODO(), nil)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	defer tx.tx.Rollback()
+
+	tasks, err := tx.GetTasksByFilter(filters)
+
+	log.Info(tasks)
+
+	jsoned, err := json.Marshal(tasks)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	writer.Write(jsoned)
+
+	return
+}
+
+func (a *AsyncTasksApp) CreateTaskRequest(writer http.ResponseWriter, r *http.Request) {
+	var rawtask AsyncTask
+
+	body, err := ioutil.ReadAll(io.LimitReader(r.Body, 104857600))
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	if err := r.Body.Close(); err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	if err := json.Unmarshal(body, &rawtask); err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	if rawtask.Type == "" {
+		errored(writer, "Task type must be provided")
+		return
+	}
+
+	for _, behavior := range rawtask.Behaviors {
+		if behavior.BehaviorType == "" {
+			errored(writer, "All behaviors must have a type")
+			return
+		}
+	}
+
+	if len(rawtask.Status) > 1 {
+		errored(writer, "A new task may only include one initial status")
+		return
+	}
+
+	if len(rawtask.Status) > 0 && rawtask.Status[0].Status == "" {
+		errored(writer, "A blank status is not allowed")
+		return
+	}
+
+	tx, err := a.db.BeginTx(context.TODO(), nil)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+	defer tx.tx.Rollback()
+
+	id, err := tx.InsertTask(rawtask)
+	if err != nil {
+		errored(writer, err.Error())
+		return
+	}
+
+	log.Info(id)
+
+	tx.tx.Commit()
+
+	url, _ := a.router.Get("getById").URL("id", id)
+	log.Info(url)
+
+	writer.Header().Set("Location", url.EscapedPath())
+	writer.WriteHeader(http.StatusCreated)
```

Comment:
Pretty interesting way to respond to a create request 👍

Reply:
It's the correct way to do it! I'm less sure about the ones adding statuses and behaviors, but since those are only fetchable as part of the overall task I figure that's still the relevant `Location` for fetching an updated representation.